### PR TITLE
feat: deprecate the `t.setPageLoadTimeout` method

### LINKED ISSUE #5868: Deprecate the 't.setPageLoadTimeout' method
The new way to set up this timeout is

```js
test
    .timeouts({ pageLoadTimeout: 10000 })
    ('test', async t => {

    });
```

### PR DESCRIPTION
**Pull request by wentwrong**
## Approach
* emit a deprecationg warning when the `t.setPageLoadTimeout` method is used
* add missed type definitions for the `Test.timeouts` method
* additional type checking for the test timeouts
## References
closes #5868

## Pre-Merge TODO
- [x] Write tests for your proposed changes
- [ ] Make sure that existing tests do not fail

### CODE REVIEW

File: src/notifications/warning-message.js [hunk 1]
```diff
@@ -47,6 +47,7 @@ export default {
...
     retryTestPagesIsNotSupported:       'Cannot enable the \'retryTestPages\' option in "{browserAlias}". Please ensure that your version of "{browserAlias}" supports the Service Worker API (https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API).\n',
     browserProviderDropOfPerformance:   'We detected \'{browserName}\' runs slowly. Try to free up or allocate more system resources on its host machine.',
-    testsCompilationTakesTooLong:       'Tests took too long to compile ({compileTime}). Ensure the test code has no excessive imports.'
+    testsCompilationTakesTooLong:       'Tests took too long to compile ({compileTime}). Ensure the test code has no excessive imports.',
+    setPageLoadTimeoutDeprecated:       'You used the \'setPageLoadTimeout\' method of TestController which is deprecated. Note that the deprecated methods will be removed in the next major release. Specify the \'pageLoadTimeout\' option using the \'Test.timeouts\' method instead.'
```
Comment (wentwrong) [inline src/notifications/warning-message.js:51]: @DIRECTcut, could you please check this warning message?

Comment (miherlosev) [inline src/notifications/warning-message.js:51]: Now, we have 2 other types of deprecation messages:
- https://github.com/DevExpress/testcafe/blob/master/src/notifications/deprecation-message.js
- https://github.com/DevExpress/testcafe/blob/master/src/configuration/deprecated-options.ts

It's time to standardize deprecation messages.

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: src/notifications/warning-message.js (hunk 1 above)
Comment (DIRECTcut) [inline src/notifications/warning-message.js:51]:
```suggestion
\'TestController.setPageLoadTimeout()\' is deprecated and will be removed in the next major release. Use \'Test.timeouts()\' to set the \'pageLoadTimeout\'.
```

Ideally (but only if it's possible to somehow highlight syntax):

```suggestion
\'TestController.setPageLoadTimeout()\' is deprecated and will be removed in the next major release. Use \'Test.timeouts()\' to set the \'pageLoadTimeout\':

test
    .timeouts({ pageLoadTimeout: 10000 })
```

BTW is `test.timeouts()` new? We don't have it documented.

Comment (wentwrong) [reply src/notifications/warning-message.js:51]: Thanks. The `test.timeouts` method is not new. It was introduced in the context of "Configure request timeouts" feature but not documented. I'll create the issue to document specifying the timeouts for an individual test after this PR will be merged.

File: test/functional/fixtures/api/es-next/page-load-timeout/test.js [hunk 2]
```diff
@@ -1,24 +1,47 @@
...
+                pageLoadTimeout: 0,
+                selectorTimeout: 10000
+            });
+        });
+
+        it('Should raise a deprecation warning if the `t.setPageLoadTimeout` method is used', function () {
+            return runTests('testcafe-fixtures/test-level.js', 'The `t.setPageLoadTimeout` method should raise a deprecation warning')
+                .then(() => {
+                    expect(testReport.warnings).eql([
+                        'The \'TestController.setPageLoadTimeout\' method is deprecated ' +
```
Comment (miherlosev) [inline test/functional/fixtures/api/es-next/page-load-timeout/test.js:40]: Add test that called the `t.setPageLoadTimeout` in loop and ensure that report contains only 1 warning message for it.

**Review by miherlosev — CHANGES_REQUESTED** [review]

**Review by miherlosev — APPROVED** [review]
Excellent

**Comment by wentwrong** [comment]
@DIRECTcut, please take a look one more time to the deprecated warning message. This is an example with used deprecated `tsConfigPath` option as CLI flag and the `t.setPageLoadTimeout` method in the test:

deprecation-warnings

**Comment by DIRECTcut** [comment]
@wentwrong 

The first one looks good to me.

The second one:
```text
The `t.setPageLoadTimeout` method is deprecated and will be removed in the next major release. Use the `test.timeouts` method to set the `pageLoadTimeout`.
```

**Review by DIRECTcut — APPROVED** [review]

**Pull request merged**